### PR TITLE
Add filter menu tests

### PR DESCRIPTION
**Pull request by huddy987**
Also adds some utilities for adding books directly through firebase (need to go through the UI to remove them still)

### CODE REVIEW

File: app/src/androidTest/java/com/example/bookwormadventuresdeluxe2/TestUtils.java [hunk 1]
```diff
@@ -125,4 +128,100 @@ public static void deleteTestBook(Solo solo, Resources resources, int title)
...
+                    "");
+            /* Set the second test account as the requested */
+            requestedBook.addRequester(resources.getString(R.string.test_account2_username));
+            borrowedBook = new Book(resources.getString(R.string.test_account1_username),
+                    resources.getString(R.string.test_book_borrowed_title),
+                    resources.getString(R.string.test_book_borrowed_author),
+                    resources.getString(R.string.test_book_borrowed_description),
+                    resources.getString(R.string.test_book_isbn),
+                    Status.Borrowed,
+                    "");
```
Comment (JawadHossain) [inline app/src/androidTest/java/com/example/bookwormadventuresdeluxe2/TestUtils.java:176]: Current Implementation keeps the borrower in the requested array and removes others for a borrowed book. 
We should add the borrower to the requester array here, right?

**Review by JawadHossain — COMMENTED** [review]
Thanks for implementing the BookManger. Please see the comment on borrowedBook

File: app/src/androidTest/java/com/example/bookwormadventuresdeluxe2/TestUtils.java (hunk 1 above)
Comment (huddy987) [reply app/src/androidTest/java/com/example/bookwormadventuresdeluxe2/TestUtils.java:177]: Done

**Review by JawadHossain — APPROVED** [review]

File: app/src/androidTest/java/com/example/bookwormadventuresdeluxe2/TestUtils.java (hunk 1 above)
Comment (IsaakC) [reply app/src/androidTest/java/com/example/bookwormadventuresdeluxe2/TestUtils.java:177]: sorry, but this should be the same for all books that are not available, eg accepted and the intermediate status ones if you add in future

**Review by IsaakC — CHANGES_REQUESTED** [review]

File: app/src/androidTest/java/com/example/bookwormadventuresdeluxe2/TestUtils.java (hunk 1 above)
Comment (huddy987) [reply app/src/androidTest/java/com/example/bookwormadventuresdeluxe2/TestUtils.java:177]: what should be the same? I'm not sure I understand what you mean

Comment (IsaakC) [reply app/src/androidTest/java/com/example/bookwormadventuresdeluxe2/TestUtils.java:177]: The accepted book should have the borrower added to the requesters list

**Review by IsaakC — APPROVED** [review]

**Pull request merged**